code/modules/space transition/space_transition.dm

```diff
@@ -0,0 +1,176 @@
+//This is realisation of the working torus-looping randomized-per-round space map, this kills the cube
+
+var/list/z_levels_list = list()
+
+/datum/space_level
+	var/name = "Your config settings failed, you need to fix this for the datum space levels to work"
+	var/list/neigbours
+	var/z_value = 1 //actual z placement
+	var/linked = 1
+	var/xi
+	var/yi   //imaginary placements on the grid
+
+/datum/space_level/New()
+	neigbours = list()
+	var/list/L = list("1","2","4","8")
```

Comment:
yeah you never actually use the directions in their number form in this code (as far as I can see, it's late) so you can just swap them for defines of strings (the default direction defines are numbers, and will make byond cry by thinking they're indexes)

eg:
```
#define Z_NEIGHBOUR_NORTH "north"
#define Z_NEIGHBOUR_EAST "east"
```
etc.